src/common/utxobased/engine/makeUtxoEngineState.ts

```diff
@@ -110,15 +110,21 @@ export function makeUtxoEngineState(
     updateTransactionCache: {}
   }
 
+  const clearObject = (obj: { [key: string]: any }): void => {
+    Object.keys(obj).forEach(key => {
+      // eslint-disable-next-line @typescript-eslint/no-dynamic-delete
+      delete obj[key]
+    })
+  }
```

Comment:
Why is this needed?
Doing something like this in Javascript is almost never needed and if it does, it usually means you are hiding some other issue.

Reply:
This is to avoid reference errors. It's a fix to the poor implementation. Each cache object should be a Map object (https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Map/clear). Though I as I understand, Maps introduce challenges on some targeted platforms.